Reporting/ad-extensions-reporter.js

```diff
@@ -0,0 +1,395 @@
+    }
+    var jsonAttributeValuesByExtensionType = {};
+    var disapprovalShortNamesByExtensionType = {};
+    var duplicateAttributeValuesCounts = {};
+
+    while (reportRows.hasNext()) {
+        var row = reportRows.next();
+
+        var placeholderTypeId = row['PlaceholderType']
+        if (extensionPlaceholderTypeIds.indexOf(placeholderTypeId) === -1) {
+            continue;
+        }
+        if (Object.keys(jsonAttributeValuesByExtensionType).indexOf(placeholderTypeId) === -1) {
+            jsonAttributeValuesByExtensionType[placeholderTypeId] = [];
+            disapprovalShortNamesByExtensionType[placeholderTypeId] = [];
+        }
+
+        var jsonAttributeValues = row['AttributeValues'];
+        var jsonDisapprovalShortNames = row['DisapprovalShortNames'];
+        if ('' === jsonDisapprovalShortNames.toString()) {
+            var disapprovalShortNames = [];
+        } else {
+            var disapprovalShortNames = JSON.parse(jsonDisapprovalShortNames);
+        };
+        if (jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues) === -1) {
+            jsonAttributeValuesByExtensionType[placeholderTypeId].push(jsonAttributeValues);
+            disapprovalShortNamesByExtensionType[placeholderTypeId].push(disapprovalShortNames);
+        } else {
+            if (Object.keys(duplicateAttributeValuesCounts).indexOf(jsonAttributeValues) === -1) {
+                duplicateAttributeValuesCounts[jsonAttributeValues] = 2;
+            } else {
+                duplicateAttributeValuesCounts[jsonAttributeValues] += 1;
+            };
+            for (var i = 0, n = disapprovalShortNames.length; i < n; i++) {
+                var disapprovalName = disapprovalShortNames[i];
+                if (disapprovalShortNamesByExtensionType[placeholderTypeId][
+                    jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues)
+                ].indexOf(disapprovalName) === -1) {
+                    disapprovalShortNamesByExtensionType[placeholderTypeId][
+                        jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues)
+                    ] = disapprovalShortNamesByExtensionType[placeholderTypeId][
+                        jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues)
+                    ].concat(disapprovalName);
+                }
+            }
+        }
+        extensionTypeFeedItemCounts[placeholderTypeId] += 1;
+    }
+
+    // Create a spreadsheet tab for each extension type, and output its data
+    Logger.log("Outputting data to spreadsheet");
+    for (var placeholderTypeId in jsonAttributeValuesByExtensionType) {
+        var placeholderTable = extensionPlaceholderTypeIdToFieldIdsObjectMapping[placeholderTypeId];
+        var orderedAttributeIds = Object.keys(placeholderTable).sort();
+        orderedAttributeIds.pop();
```

Comment:
Does this remove the name? maybe worth adding in a comment saying that is what you are doing? As without remembering the name is the first element, and isn't relevant for the rest of this code, you might not know!

Reply:
Yes it does. I agree this should be clearer and I've added a comment. After sorting, `'name'` ends up at the end of the array, which is why `.pop()` works. I've made it more explicit now by using `.splice()` instead.